java-engine/src/main/java/io/getunleash/engine/UnleashEngine.java

```diff
@@ -66,21 +63,25 @@ public void countVariant(String flagName, String variantName) {
     }
 
     public MetricsBucket getMetrics() throws YggdrasilError {
-        YggResponse<MetricsBucket> response = read(yggdrasil.getMetrics(), new TypeReference<>() {});
+        YggResponse<MetricsBucket> response = read(yggdrasil.getMetrics(), new TypeReference<>() {
+        });
         return response.getValue();
     }
 
     /**
      * Handle reading from a pointer into a String and mapping it to an object
      */
-    private <T> T read(Pointer pointer, TypeReference<T> typeReference) {
-        String str = pointer.getString(0, UTF_8);
-        yggdrasil.freeResponse(pointer);
+    private <T> T read(Pointer pointer, TypeReference<T> clazz) {
         try {
-            System.out.println(str); // TODO use a logging library. SLF4J?
-            return reader.forType(typeReference).readValue(str);
-        } catch (IOException e) {
-            throw new YggdrasilParseException(str, typeReference.getClass(), e);
+            String str = pointer.getString(0, UTF_8);
+            try {
+                return mapper.readValue(str, clazz);
+            } catch (IOException e) {
+                System.out.println("Failed to parse response from Yggdrasil: " + str);
```

Comment:
Can we trim this? Certainly for benchmarks, streaming to std out is costly. I removed this line when I ran the original benchmarks
```suggestion

```

Reply:
Yes, I'd like to integrate with SLF4J and add this as debug log, but I'd like to keep it for now to later replace all System.out with the logger

Reply:
Fair enough, but you'll need to trim this out for your benchmark runs to get anything honest

Reply:
Yes, the logger will just be disabled for benchmarks